### ISSUE TITLE
Particle system one shot issues

### ISSUE THREAD

**Issue by Dusk-Dawn** [godotengine/godot#83599]
### Godot version

4.1.1

### System information

Godot 4.1.1

### Issue description

GPU Particle 2D emit particles only once when one shot is true.
after the first emission you can't force it to emit again unless you use restart(), but that has problems, mainly the fact it kills the current particles playing.

Also when you set emitting = true in any particle system it does not "starts" the emission it "resume" it. that's not correct, I believe it SHOULD indeed start emitting anew.

being able to emit a "burst of particles" without the need to instantiate a new particle system is a huge example of good particles usage, and it's currently lacking in Godot
I feel the particle system is lacking common methodes such as a "simple emit x amount", play once, pause, kill all particles alive, etc

### Steps to reproduce

try emitting again after emitting once if one shot is enabled without using .restart()
or try to emit any particle again right after it already emited ( set amount to 1 and life time to 5 to see it more clearly)

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
Can you please try this with a supported version, like 4.1.2 (only the latest patch version is supported, and it might already be fixed)

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot/issues/17903 (possible duplicate).

Remember that you can use manual particle emission in 4.x using a custom shader. 4.2.beta2 also has an Amount Ratio property to adjust the number of particles emitted without restarting the particle system.

**Comment by Dusk-Dawn** [comment]
no change in 4.1.2

I'm trying to port a project that used A LOT of those particle bursts to quickly and very efficiently display LOTS of effects.
I even made singleton particles and turned it into a pattern that I used in all my projects for performance and ease of use

**Comment by clayjohn** [comment]
This isn't a bug. Once you emit a one shot particle system, you need to wait until the particle system stops processing before toggling ``emitting`` will restart the particles. 

I agree the behaviour is a bit awkward. We have a proposal to track discussion and come up with a better alternative. https://github.com/godotengine/godot-proposals/issues/7322

In 4.2 there is a better option, if you want to emit the particles again as soon as possible. You should emit the particles after receiving the [``finished``](https://docs.godotengine.org/en/latest/classes/class_gpuparticles2d.html#signals) signal.

Since this is a duplicate of https://github.com/godotengine/godot/issues/79689 and better tracked by https://github.com/godotengine/godot-proposals/issues/7322 I will be closing this issue. If you have a suggestion, please add it to https://github.com/godotengine/godot-proposals/issues/7322

**Comment by OI3AKE** [comment]
Random User here,
for those who still have the same issue attach this thing to your Particle (made for CPU you can edit it to work with GPU particle as well.

(Notice for people who use 4.3+ on future, the code is for 4.3 idk if any method or property name changed or not.)

```swift
@tool
extends CPUParticles3D

@export var oneshoot_enabled = false

func _process(delta: float) -> void:
	if self.emitting:
		await get_tree().create_timer(lifetime - 0.5).timeout
		self.emitting = false
```

**Comment by AThousandShips** [comment]
This belongs in the proposal that's linked above, this is a duplicate

**Comment by OI3AKE** [comment]
i know
sorry but you can share this to the original post as well as some people might not know this strange behavior with the particle thing
